src/Analyzer/FunctionNode.cpp

```diff
@@ -166,7 +166,12 @@ ASTPtr FunctionNode::toASTImpl() const
     auto function_ast = std::make_shared<ASTFunction>();
 
     function_ast->name = function_name;
-    function_ast->is_window_function = isWindowFunction();
+
+    if (isWindowFunction())
+    {
+        function_ast->is_window_function = true;
```

Comment:
now that we have `kind`, we can remove `is_window_function`. Also maybe we can call it `FunctionType` and `EngineType` or if you have any other suggestions , what do you think ?

Reply:
For me `kind` feels a bit less confusing because `Type` could mean C++ type defined with typedef of using.
I'll remove `is_window_function` as you said but I'd prefer to do that in a separate PR.